src/main/drivers/timer_stm32f10x.c

```diff
@@ -23,25 +23,56 @@
 #include "rcc.h"
 #include "timer.h"
 
-const timerDef_t timerDefinitions[HARDWARE_TIMER_DEFINITION_COUNT] = {
+#define TIM_N(n) (1 << (n))
+#define USED_TIMER_COUNT BITCOUNT(USED_TIMERS)
+
+const timerDef_t timerDefinitions[USED_TIMER_COUNT] = {
+#if USED_TIMERS & TIM_N(1)
     { .TIMx = TIM1,  .rcc = RCC_APB2(TIM1), .inputIrq = TIM1_CC_IRQn },
```

Comment:
Macro may be used here, something like (untested):
```
#define TIM_DEFINITION(n, rcc, irq) { .TIMx = CONCAT(TIM, n), .rcc = rcc, .inputIrq = irq }
...
#if USED_TIMERS & TIM_N(1)
  TIM_DEFINITION(1, RCC_APB2(TIM1), TIM1_CC_IRQn),
#endif
```
`rcc` may be further modified to accept only APB2 as parameter, but macro expansion rules can easily make this problematic (TIM1 must be concatenated directly, or it will expand to TIM1 address too soon)

Adding new members would be much easier than (timer number would be trivial, but structure packing should be considered to save space).
 Also tricks from timer channel DMA 'database' can be used (if it is worth the confusion)